test/routes/home_route_test.dart

```diff
@@ -13,53 +13,75 @@ void main() {
-      permissionsObserver.setPermissionResolvable(Permission.storage, () => permissionGrantCompleter.future);
-      await tester.tap(find.text(l10n.grant));
-      expect(permissionsObserver.requestedPermissions, [Permission.storage]);
-      await tester.pump();
-      expect(find.byType(CircularProgressIndicator), findsOneWidget,
-          reason: 'Indicate while waiting for the permission to be granted');
-      permissionGrantCompleter.complete(PermissionStatus.granted);
-      await tester.pumpAndSettle();
-      expect(find.byType(Home), findsOneWidget);
+  group('permissions screen', () {
+    testWidgets('shows if no permissions were granted and pressing the button requests permissions',
+            (WidgetTester tester) async {
+      late PermissionsChannelObserver permissionsObserver;
+      await setUpAppTest(() {
+        permissionsObserver = PermissionsChannelObserver(tester.binding);
+        permissionsObserver.setPermission(Permission.storage, PermissionStatus.denied);
+        permissionsObserver.setPermission(Permission.audio, PermissionStatus.denied);
+      });
+      await tester.runAppTest(() async {
+        expect(permissionsObserver.checkedPermissions, [Permission.storage, Permission.audio],
+            reason: 'Should always check the storage and audio permission on startup');
+        expect(find.byType(Home), findsNothing, reason: 'Permissions are not granted yet');
+        final permissionGrantCompleter = Completer<PermissionStatus>();
+        permissionsObserver.setPermissionResolvable(Permission.storage, () => permissionGrantCompleter.future);
+        permissionsObserver.setPermissionResolvable(Permission.audio, () => permissionGrantCompleter.future);
+        await tester.tap(find.text(l10n.grant));
+        expect(permissionsObserver.requestedPermissions, [Permission.storage, Permission.audio]);
+        await tester.pump();
+        expect(find.byType(CircularProgressIndicator), findsOneWidget,
+            reason: 'Indicate while waiting for the permission to be granted');
+        permissionGrantCompleter.complete(PermissionStatus.granted);
+        await tester.pumpAndSettle();
+        expect(find.byType(Home), findsOneWidget);
+      });
     });
-  });
 
-  testWidgets('permissions screen - shows toast and opens settings when permissions are denied',
-      (WidgetTester tester) async {
-    late PermissionsChannelObserver permissionsObserver;
-    await setUpAppTest(() {
-      permissionsObserver = PermissionsChannelObserver(tester.binding);
-      permissionsObserver.setPermission(Permission.storage, PermissionStatus.denied);
+    // On Android 13+ the storage permission is removed and reports as permanentlyDenied.
+    // If the user granted audio permissions, we have access to the music
+    // and don't want to show the permission request screen.
+    testWidgets('does not show when one removed permission is permanently denied', (WidgetTester tester) async {
```

Comment:
why not test it for both?
on API `<33` he `Permission.audio` is permantently denied